parse.py

```diff
@@ -345,23 +361,103 @@ def check_birth_after_parent_marriage(families, individuals):
         #if "MARR" in families[id]:
         if get_family_data_by_key(families,id,"MARR"):
             marriageDate = datetime.datetime.strptime(get_family_data_by_key(families,id,"MARR"), "%d %b %Y").date()
-            for childID in get_family_data_by_key(families,id,"CHIL"):
-                childBirthDate = datetime.datetime.strptime(get_family_data_by_key(families,id,"MARR"), "%d %b %Y").date()
+            childID = get_family_data_by_key(families,id,"CHIL")
+            if childID:
+                childBirthDate = datetime.datetime.strptime(get_individual_data_by_key(individuals,childID,"DATE"), "%d %b %Y").date()
                 if childBirthDate < marriageDate:
                     print("ANOMALY: FAMILY: US08: Child ("
-                            + get_individual_name(childID, individuals).replace("/", "")
+                            + childID
                             + ") born before marriage of parents in family: "
                             + id
                             + ".")
                     is_valid = False
         else:
-            print(families[id])
             print("ERROR: FILE: US08: Marriage date not set or properly formatted of family: "
                 + id
                 + ".")
             is_valid = False
     return is_valid
 
+#US07
+def check_age(birth_date):
+    today = datetime.datetime.today()
+    birth = datetime.datetime.strptime(birth_date, "%d %b %Y")
+    age_from_today = today - birth
+    age = age_from_today.days
+    return age
+
+def check_age_under_onefifty(individuals):
+    is_valid = True
+    birth_date = 0
+    for id in individuals:
+        individual = individuals[id]
+        birthday = get_individual_data_by_key(individuals,id,"DATE")
+        day_age = check_age(birthday)
+        age = day_age/365.25
+        if age >= 150:
+            is_valid = False
+            print("Age is over 150")
+    return is_valid    
+
+#US21
+def check_gender_role(families, individuals):
+    is_valid = True
+    for id in families:
+        husbID = get_family_data_by_key(families,id,"HUSB")
+        if husbID:
+            Gender = get_individual_data_by_key(individuals,husbID,"SEX")
```

Comment:
Can we follow variable naming convention, and change Gender to gender please?

Reply:
Addressed in US34-US35